File: 1402-reducing-dishes/1402-reducing-dishes.cpp
using namespace std;

class Solution {
public:
    int maxSatisfaction(vector<int>& satisfaction) {
        sort(satisfaction.begin(), satisfaction.end(), greater<int>());
        long long int s=0, ans=0, pointer=0;
        while(pointer<satisfaction.size() && s+satisfaction[pointer]>0){
            s+=satisfaction[pointer];
            ans+=s;
            pointer++; 
        }
        return ans;
    }
};

File: 1402-reducing-dishes/1402-reducing-dishes.py
class Solution:
    def maxSatisfaction(self, satisfaction: List[int]) -> int:
        s = 0
        ans = 0
        satisfaction.sort(reverse=True)
        pointer = 0
        while pointer < len(satisfaction):
            if s + satisfaction[pointer] <= 0:
                return ans
            else:
                s+=satisfaction[pointer]
                ans+=s
            pointer += 1
        return ans